index.js

```diff
@@ -101,7 +82,7 @@ function getConstructorName(errorLike) {
   var constructorName = errorLike;
   if (errorLike instanceof Error) {
     constructorName = getFunctionName(errorLike.constructor);
-  } else if (typeof errorLike === 'function') {
+  } else if (type(errorLike) === 'function') {
```

Comment:
Similarly here - `typeof` would be better, as we closely control the flow here - `type-detect` is better put to use where we want to check object types. If we're only ever checking primitives, it's worth considering avoiding using `type-detect` entirely IMO